tests/core/data/test_data_node.py

```diff
@@ -186,6 +186,53 @@ def test_read_write(self):
         assert dn.is_ready_for_reading
         assert dn.job_ids == [job_id]
 
+    def test_lock_unlock_dn(self):
+        dn = InMemoryDataNode("dn", Scope.SCENARIO)
+        assert not dn.edit_in_progress
+        assert dn._editor_id is None
+        assert dn._editor_expiration_date is not None
+
+        dn.lock_edit("user_1")
+        assert dn.edit_in_progress
+        assert dn._editor_id == "user_1"
+        assert dn._editor_expiration_date is not None
+        with pytest.raises(DataNodeIsBeingEdited):
+            dn.lock_edit("user_2")
+        with pytest.raises(DataNodeIsBeingEdited):
+            dn.unlock_edit("user_2")
+        dn.unlock_edit("user_1")
+        assert not dn.edit_in_progress
+        assert dn._editor_id is None
+        assert dn._editor_expiration_date is not None
+
+        dn.lock_edit("user_2")
+        assert dn.edit_in_progress
+        assert dn._editor_id == "user_2"
+        assert dn._editor_expiration_date is not None
+        dn.lock_edit()
+        assert dn.edit_in_progress
+        assert dn._editor_id is None
+        assert dn._editor_expiration_date is not None
+        dn.unlock_edit()
```

Comment:
Note that I changed the last line of the test.
The assert does not bring test value to me.
I would change the implementation to change the test as well.

```suggestion
    def test_none_editor_can_lock_a_locked_dn(self):
        dn = InMemoryDataNode("dn", Scope.SCENARIO)
        dn.lock_edit("user")
        dn.lock_edit()
        assert dn.edit_in_progress
        assert dn._editor_id is None
        assert dn._editor_expiration_date is None
```